cli/tests/program.rs

```diff
@@ -1229,12 +1229,29 @@ fn test_cli_program_set_buffer_authority() {
         panic!("not a buffer account");
     }
 
-    // Set authority to buffer
+    // Attempt to deploy program from buffer using previous authority (should fail)
+    config.signers = vec![&keypair, &buffer_keypair];
+    config.command = CliCommand::Program(ProgramCliCommand::Deploy {
+        program_location: Some(noop_path.to_str().unwrap().to_string()),
+        program_signer_index: None,
+        program_pubkey: None,
+        buffer_signer_index: None,
+        buffer_pubkey: Some(buffer_keypair.pubkey()),
+        allow_excessive_balance: false,
+        upgrade_authority_signer_index: 0,
+        is_final: false,
+        max_len: None,
+        skip_fee_check: false,
+    });
+    config.output_format = OutputFormat::JsonCompact;
+    process_command(&config).unwrap_err();
+
+    // Set buffer authority back to original authority
     config.signers = vec![&keypair, &new_buffer_authority];
     config.command = CliCommand::Program(ProgramCliCommand::SetBufferAuthority {
         buffer_pubkey: buffer_keypair.pubkey(),
         buffer_authority_index: Some(1),
-        new_buffer_authority: buffer_keypair.pubkey(),
+        new_buffer_authority: keypair.pubkey(),
```

Comment:
Changing `buffer_keypair ` -> `keypair` here to make it a bit more clear (separate keys serving as **authorities** from keys serving as buffers/programs/etc).

Reply:
It was deliberate to set the authority to the buffer itself, as evidenced by the comment `// Set authority to buffer`. There are often programmer bugs if someone tries this behavior, and we need to support it, which is why the test is there. Can you revert that part back?

Reply:
Interesting, reverted in https://github.com/solana-labs/solana/pull/34316/commits/71972c2545df46e6ab1df387c71fad91715bc281 (also added some clarifying comments).